Updated contents and structure (#97)
matter-labs/zksync-web-v2-docs: docs/dev/README.md

# zkSync Documentation

This documentation is intended to assist you in developing on zkSync. 
It introduces the concept of zkSync, describes the zkSync technology stack, and some advanced subjects for complex applications and use cases.

> Because this is a community initiative, feel free to suggest new topics, add new content, and provide examples wherever you believe they will be useful. If you're unsure how to proceed, [follow these instructions](#).

Author: Yeah working on the page.

> Because this is a community initiative, feel free to suggest new topics, add new content, and provide examples wherever you believe they will be useful. If you're unsure how to proceed, [follow these instructions](#).

Author: Open sourced.

## Fundamental topics

If this is your first time using zkSync, we recommend that you kickoff at the beginning and work your way through the documentation like you would with a book.

- [Intro to Rollups](./fundamentals/rollups.md) - A brief overview of rollups.

> - [zkSync overview](./fundamentals/zkSync.md) - A quick overview of zkSync technologies.

Reviewer: ```suggestion
- [zkSync overview](./fundamentals/zkSync.md) - A quick overview of zkSync technology.
```

Reviewer 2: technology? network/rollup or even system sounds more appropriate

- [zkSync testnet](./fundamentals/testnet.md) - A brief intro into zkSync testnest.

>   - [migration-testnet](./testnet/migration-to-testnet-paymaster.md) - A brief explainer on paymasters on testnet.

Reviewer: > migration-testnet

Is it intended?

The link is also mostly about how to migrate to the paymasters for a person who was using "paying fees with ERC20 tokens" feature of the previous testnets on zkSync. I think that a brief explainer is here: https://aqwzx-zksync-v2-docs--pr97-roadmap-1-qia5fdp3.web.app/dev/developer-guides/transactions/aa.html#paymasters

 
### Developer guides

- [Quickstart](./developer-guides/hello-world.md) - Learn how to build a full dApp using the zkSync development toolbox.

> - [Contracts](#) - An overview on zkSync system contracts and smart contracts.

Reviewer: Is it intended to have links that don't lead anywhere?

>   - [System contract](./developer-guides/contracts/system-contracts.md) - A brief overview of zkSync contract.

Reviewer 2: contracts? system contracts?

  - [Contract deployment](./developer-guides/contracts/contracts.md) - A guide on how to deploy smart contract on zkSync.

>   - [Contract verification](./developer-guides/contracts/contract-verification.md) - A guide on how to verify smart contract with zkSync block explorer.

Reviewer: ```suggestion
  - [Contract verification](./developer-guides/contracts/contract-verification.md) - A guide on how to verify smart contract on zkSync block explorer.
```

- [Transactions](#) - Learn how zkSync handles transactions.
  - [Transaction finality](./developer-guides/transactions/transactions.md) - A quick guide on zkSync transaction finality.

>   - [Account abstraction](./developer-guides/transactions/aa.md) - Learn how zkSync 2.0 can initiate transactions, like an EOA.

Reviewer: zkSync 2.0 is a platform and it does not initiate transactions :)

>   - [Account abstraction](./developer-guides/transactions/aa.md) - Learn how zkSync 2.0 can initiate transactions, like an EOA.

Reviewer 2: how to initiate transactions on zksync?

- [Fee mechanism](./developer-guides/fee-model.md) - A quick overview of zkSync fee structure.

> - [Bridging of funds](./developer-guides/Bridging/bridging-funds.md) - A brief intro on funds bridging.

Reviewer 2: funds --> assets/token?

  - [L1 / L2 Interoperability](./developer-guides/Bridging/l1-l2-interop.md) - A quick brief on data communication between l1 and l2.
    - [L1 / L2 communication](./developer-guides/Bridging/l1-l2.md) - Learn how to send data from Ethereum to zkSync.
    - [L2 / L1 communication](./developer-guides/Bridging/l2-l1.md) - Learn how to send data from zkSync to Ethereum.
- [Troubleshooting](#)
  - [Reporting issues](./troubleshooting/reporting-issues.md) - Report known issues to us here.
  - [Important links](./troubleshooting/important-links.md) - Get quick reference to important links.
  - [Temporary limitations](./troubleshooting/temp-limits.md)

>   - [Status](./troubleshooting/status.md)

Reviewer 2: maybe add some description here/

>   - [FAQ's](./troubleshooting/faq/known-issues.md) - Get answers to your frequently asked questions

Reviewer 2: period missing in the end

> - [Glossary](./glossary/glossary.md) - An alphabetical list of words relating to zkSync and it's ecosystem.

Reviewer 2: relating -> related ?

Author: yes

### Developer Tools

- Faucet, bridge, wallet - Explore our faucet, bridge and wallet feature.
[zkSync 2.0 Portal](https://portal.zksync.io)

> - Block explorer - Search for real-time and historical information about a blocks, transactions, addresses, and more on zkSync.

Reviewer: ```suggestion
- Block explorer - Search for real-time and historical information about blocks, transactions, addresses, and more on zkSync.
```

> - Block explorer - Search for real-time and historical information about a blocks, transactions, addresses, and more on zkSync.

Reviewer 2: remove 'a', no need for the last period

[Explorer](https://zksync2-testnet.zkscan.io)

### Examples and tutorials

> - [Cross-chain governance](./tutorials/cross-chain-tutorial.md) - Learn how to implement L1 to L2 contract interaction.

Reviewer 2: L1-to-L2

> - [Account abstraction](./tutorials/custom-aa-tutorial.md) - Learn how to deploy your custom accounts and interact directly with the zkSync system contract.

Reviewer: ```suggestion
- [Account abstraction](./tutorials/custom-aa-tutorial.md) - Learn how to deploy your custom accounts and interact zkSync system contracts.
```

> - [Building custom paymasters](./tutorials/custom-paymaster-tutorial.md) - Learn how to build a custom paymaster to allow users pay fees in our token.

Reviewer: ```suggestion
- [Building custom paymasters](./tutorials/custom-paymaster-tutorial.md) - Learn how to build a custom paymaster to allow users to pay fees in your token.
```

- [zkSync overview](./fundamentals/zkSync.md) - A quick overview of zkSync technologies.
- [zkSync testnet](./fundamentals/testnet.md) - A brief intro into zkSync testnest.
 
### Developer guides

- [Quickstart](./developer-guides/hello-world.md) - Learn how to build a full dApp using the zkSync development toolbox.
- [Contracts](./developer-guides/contracts/system-contracts.md) - A brief overview of zkSync system contract.
  - [Contract deployment](./developer-guides/contracts/contracts.md) - A guide on how to deploy smart contract on zkSync.
  - [Contract verification](./developer-guides/contracts/contract-verification.md) - A guide on how to verify smart contract with zkSync block explorer.
- [Transactions](./developer-guides/transactions/transactions.md) - A guide on how zkSync handles transactions.
  - [Account abstraction](./developer-guides/transactions/aa.md) - Learn how zkSync 2.0 platform handles transactions.
  - [Blocks](./developer-guides/transactions/blocks.md) - Learn how blocks work on zkSync.
  - [Account](./developer-guides/transactions/account.md) - Learn how accounts work on zkSync.
=======

>   - [Contract verification](./developer-guides/contracts/contract-verification.md) - A guide on how to verify smart contract on zkSync block explorer.

Author: What are you referring to?

  - [L1 / L2 Interoperability](./developer-guides/bridging/l1-l2-interop.md) - A quick brief on data communication between l1 and l2.
    - [L1 / L2 communication](./developer-guides/bridging/l1-l2.md) - Learn how to send data from Ethereum to zkSync.
    - [L2 / L1 communication](./developer-guides/bridging/l2-l1.md) - Learn how to send data from zkSync to Ethereum.
    - 
- [Troubleshooting](./troubleshooting/important-links.md) - Get quick reference to important links and report known issues to us here.
  - [Status](./troubleshooting/status.md) - Get updates on things we are currently working on.
  - [FAQ's](./troubleshooting/faq/known-issues.md) - Get answers to your frequently asked questions.

- [Glossary](./glossary/glossary.md) - An alphabetical list of words relating to zkSync and it's ecosystem.

### Developer Tools

- Faucet, bridge, wallet - Explore our faucet, bridge and wallet feature.
[zkSync 2.0 Portal](https://portal.zksync.io)
- Block explorer - Search for real-time and historical information about blocks, transactions, addresses, and more on zkSync.

[Explorer](https://zksync2-testnet.zkscan.io)

### Examples and tutorials

> - [Cross-chain governance](./tutorials/cross-chain-tutorial.md) - Learn how to implement L1 to L2 contract interaction.

Reviewer: Why did you mark it as resolved?
